src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -162,20 +162,23 @@ public void Dispose()
             {
                 throw new ArgumentNullException("role");
             }
-
             var result = await ValidateRoleInternal(role);
             if (!result.Succeeded)
             {
                 return result;
             }
             await UpdateNormalizedRoleNameAsync(role);
-            return await LogResultAsync(await Store.CreateAsync(role, CancellationToken), role);
+            result = await Store.CreateAsync(role, CancellationToken);
+            using (IdentityLogger.BeginScope(await GetScopeMessageforRoleAsync(role)))
```

Comment:
Sure we already have a overload method in each manager class to create the scope message. I can refactor to have something like this. I added this in the IdentityLogger class so that it can be shared among them. Either way is fine. I'll see make the change and see how it looks